pkg/utils/kubernetes/plant.go

```diff
@@ -0,0 +1,138 @@
+// The transformation function is applied to the current state of the Plant object. If the transformation
+// yields a semantically equal Plant (regarding conditions), no update is done and the operation returns normally.
+func TryUpdatePlantConditions(g gardencore.Interface, backoff wait.Backoff, meta metav1.ObjectMeta, transform func(plant *gardencorev1alpha1.Plant) (*gardencorev1alpha1.Plant, error)) (*gardencorev1alpha1.Plant, error) {
+	return tryUpdatePlant(g, backoff, meta, transform, func(g gardencore.Interface, plant *gardencorev1alpha1.Plant) (*gardencorev1alpha1.Plant, error) {
+		return g.CoreV1alpha1().Plants(plant.Namespace).UpdateStatus(plant)
+	}, func(cur, updated *gardencorev1alpha1.Plant) bool {
+		return equality.Semantic.DeepEqual(cur.Status.Conditions, updated.Status.Conditions)
+	})
+}
+
+// TryUpdatePlantStatusWithEqualFunc tries to update the status of the Plant matching the given <meta>.
+// It retries with the given <backoff> characteristics as long as it gets Conflict errors.
+// The transformation function is applied to the current state of the Plant object. If the equal
+// func concludes a semantically equal Plant, no update is done and the operation returns normally.
+func TryUpdatePlantStatusWithEqualFunc(g gardencore.Interface, backoff wait.Backoff, meta metav1.ObjectMeta, transform func(*gardencorev1alpha1.Plant) (*gardencorev1alpha1.Plant, error), equal func(cur, updated *gardencorev1alpha1.Plant) bool) (*gardencorev1alpha1.Plant, error) {
+	return tryUpdatePlant(g, backoff, meta, transform, func(g gardencore.Interface, plant *gardencorev1alpha1.Plant) (*gardencorev1alpha1.Plant, error) {
+		return g.CoreV1alpha1().Plants(plant.Namespace).UpdateStatus(plant)
+	}, equal)
+}
+
+// TryUpdatePlantWithEqualFunc tries to update  the Plant matching the given <meta>.
+// It retries with the given <backoff> characteristics as long as it gets Conflict errors.
+// The transformation function is applied to the current state of the Plant object. If the equal
+// func concludes a semantically equal Plant, no update is done and the operation returns normally.
+func TryUpdatePlantWithEqualFunc(g gardencore.Interface, backoff wait.Backoff, meta metav1.ObjectMeta, transform func(*gardencorev1alpha1.Plant) (*gardencorev1alpha1.Plant, error), equal func(cur, updated *gardencorev1alpha1.Plant) bool) (*gardencorev1alpha1.Plant, error) {
+	return tryUpdatePlant(g, backoff, meta, transform, func(g gardencore.Interface, plant *gardencorev1alpha1.Plant) (*gardencorev1alpha1.Plant, error) {
+		return g.CoreV1alpha1().Plants(plant.Namespace).Update(plant)
+	}, equal)
+}
+
+// CreateOrPatchPlant either creates the object or patches the existing one with the strategic merge patch type.
+func CreateOrPatchPlant(g gardencore.Interface, meta metav1.ObjectMeta, transform func(*gardencorev1alpha1.Plant) *gardencorev1alpha1.Plant) (*gardencorev1alpha1.Plant, error) {
```

Comment:
Use the controller runtime `controllerutil.CreateOrUpdate` instead of implementing this on your own.